### ISSUE TITLE
Can not connect to kernel 

### ISSUE THREAD

**Issue by CPL2023** [spyder-ide/spyder#21119]
### What steps will reproduce the problem?

I keep my v2ray software open, that's the reason why i cannot use the spyder

### Traceback
```python-traceback
Exception in thread Thread-46:
Traceback (most recent call last):
  File "C:\Users\Fred\anaconda3\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 112, in run
    loop.run_until_complete(self._async_run())
  File "C:\Users\Fred\anaconda3\lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
  File "C:\Users\Fred\anaconda3\lib\asyncio\futures.py", line 201, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "C:\Users\Fred\anaconda3\lib\asyncio\tasks.py", line 232, in __step
    result = coro.send(None)
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 146, in _async_run
    self._create_socket()
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 102, in _create_socket
    self.socket = self.context.socket(zmq.REQ)
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\sugar\context.py", line 259, in socket
    s: ST = self._socket_class(  # set PYTHONTRACEMALLOC=2 to get the calling frame
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\_future.py", line 221, in __init__
    super().__init__(context, socket_type, **kwargs)  # type: ignore
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\sugar\socket.py", line 93, in __init__
    super().__init__(*a, **kw)
  File "zmq\backend\cython\socket.pyx", line 330, in zmq.backend.cython.socket.Socket.__init__
zmq.error.ZMQError: Too many open files
Exception in thread Thread-50:
Traceback (most recent call last):
  File "C:\Users\Fred\anaconda3\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 112, in run
    loop.run_until_complete(self._async_run())
  File "C:\Users\Fred\anaconda3\lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
  File "C:\Users\Fred\anaconda3\lib\asyncio\futures.py", line 201, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "C:\Users\Fred\anaconda3\lib\asyncio\tasks.py", line 232, in __step
    result = coro.send(None)
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 146, in _async_run
    self._create_socket()
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 102, in _create_socket
    self.socket = self.context.socket(zmq.REQ)
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\sugar\context.py", line 259, in socket
    s: ST = self._socket_class(  # set PYTHONTRACEMALLOC=2 to get the calling frame
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\_future.py", line 221, in __init__
    super().__init__(context, socket_type, **kwargs)  # type: ignore
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\sugar\socket.py", line 93, in __init__
    super().__init__(*a, **kw)
  File "zmq\backend\cython\socket.pyx", line 330, in zmq.backend.cython.socket.Socket.__init__
zmq.error.ZMQError: Too many open files
Exception in thread Thread-58:
Traceback (most recent call last):
  File "C:\Users\Fred\anaconda3\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 112, in run
    loop.run_until_complete(self._async_run())
  File "C:\Users\Fred\anaconda3\lib\site-packages\nest_asyncio.py", line 90, in run_until_complete
    return f.result()
  File "C:\Users\Fred\anaconda3\lib\asyncio\futures.py", line 201, in result
    raise self._exception.with_traceback(self._exception_tb)
  File "C:\Users\Fred\anaconda3\lib\asyncio\tasks.py", line 232, in __step
    result = coro.send(None)
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 146, in _async_run
    self._create_socket()
  File "C:\Users\Fred\anaconda3\lib\site-packages\jupyter_client\channels.py", line 102, in _create_socket
    self.socket = self.context.socket(zmq.REQ)
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\sugar\context.py", line 259, in socket
    s: ST = self._socket_class(  # set PYTHONTRACEMALLOC=2 to get the calling frame
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\_future.py", line 221, in __init__
    super().__init__(context, socket_type, **kwargs)  # type: ignore
  File "C:\Users\Fred\anaconda3\lib\site-packages\zmq\sugar\socket.py", line 93, in __init__
    super().__init__(*a, **kw)
  File "zmq\backend\cython\socket.pyx", line 330, in zmq.backend.cython.socket.Socket.__init__
zmq.error.ZMQError: Too many open files
```

## Versions

* Spyder version: 5.4.1  (conda)
* Python version: 3.10.9 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<9.0.0       :  8.10.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.17.3 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.5.4 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
paramiko >=2.4.0              :  2.8.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<3.0           :  2.16.2 (OK)
pylint_venv >=2.1.1           :  2.3.0 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.7.0;<1.8.0          :  1.7.1 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.2 (OK)
qstylizer >=0.2.2             :  0.2.2 (OK)
qtawesome >=1.2.1             :  1.2.2 (OK)
qtconsole >=5.4.0;<5.5.0      :  5.4.0 (OK)
qtpy >=2.1.0                  :  2.2.0 (OK)
rtree >=0.9.7                 :  1.0.1 (OK)
setuptools >=49.6.0           :  65.6.3 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.4.1;<2.5.0 :  2.4.1 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=22.1.0                  :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  3.7.0 (OK)
numpy >=1.7                   :  1.23.5 (OK)
pandas >=1.1.1                :  1.5.3 (OK)
scipy >=0.17.0                :  1.10.0 (OK)
sympy >=0.7.3                 :  1.11.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @CPL2023, thanks for reporting. This problem could be fixed in our latest version available in Anaconda (5.4.3). So, please update by closing Spyder, opening the Anaconda Prompt and running there the commands mentioned in [our documentation](http://docs.spyder-ide.org/current/installation.html#updating-with-anaconda).

Let us know if it works for you.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
